From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
Date: Wed, 7 May 2025 16:15:37 +0000
Subject: [PATCH] initial v8 patchset for wasm in interpreter mode support

Wasm in interpreter mode is implemented via DrumBrake, maintained by the
Microsoft Edge DrumBrake WebAssembly developer team.
---
 ...e-Android-in-supported-platform-for-.patch | 22 +++++++++++++++
 ...e-support-for-32-bit-binary-compilat.patch | 28 +++++++++++++++++++
 2 files changed, 50 insertions(+)
 create mode 100644 vanadium/subprojects_patches/v8/0001-drumbrake-Include-Android-in-supported-platform-for-.patch
 create mode 100644 vanadium/subprojects_patches/v8/0002-drumbrake-disable-support-for-32-bit-binary-compilat.patch

diff --git a/vanadium/subprojects_patches/v8/0001-drumbrake-Include-Android-in-supported-platform-for-.patch b/vanadium/subprojects_patches/v8/0001-drumbrake-Include-Android-in-supported-platform-for-.patch
new file mode 100644
index 0000000000000..e44dadc838da0
--- /dev/null
+++ b/vanadium/subprojects_patches/v8/0001-drumbrake-Include-Android-in-supported-platform-for-.patch
@@ -0,0 +1,22 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
+Date: Wed, 7 Aug 2024 04:55:08 +0000
+Subject: [PATCH] drumbrake: Include Android in supported platform for wasm
+ interpreter
+
+---
+ gni/v8.gni | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/gni/v8.gni b/gni/v8.gni
+index 3c4f50334fd..37fc077d332 100644
+--- a/gni/v8.gni
++++ b/gni/v8.gni
+@@ -326,6 +326,7 @@ is_drumbrake_supported =
+     v8_enable_webassembly && v8_enable_pointer_compression &&
+     (v8_current_cpu == "x64" || v8_current_cpu == "arm64") &&
+     (target_os == "win" || target_os == "linux" || target_os == "mac" ||
++     target_os == "android" ||
+      target_os == "ios")
+ 
+ # Turbofan is enabled by default, except in lite mode.
diff --git a/vanadium/subprojects_patches/v8/0002-drumbrake-disable-support-for-32-bit-binary-compilat.patch b/vanadium/subprojects_patches/v8/0002-drumbrake-disable-support-for-32-bit-binary-compilat.patch
new file mode 100644
index 0000000000000..a69e1dc1c26dc
--- /dev/null
+++ b/vanadium/subprojects_patches/v8/0002-drumbrake-disable-support-for-32-bit-binary-compilat.patch
@@ -0,0 +1,28 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
+Date: Wed, 7 Aug 2024 04:53:11 +0000
+Subject: [PATCH] drumbrake: disable support for 32-bit binary compilation
+
+---
+ BUILD.gn | 8 ++++++++
+ 1 file changed, 8 insertions(+)
+
+diff --git a/BUILD.gn b/BUILD.gn
+index 1bb7fc93c10..a37eba36c6b 100644
+--- a/BUILD.gn
++++ b/BUILD.gn
+@@ -552,6 +552,14 @@ if (v8_enable_snapshot_native_code_counters == "") {
+   v8_enable_snapshot_native_code_counters = v8_enable_debugging_features
+ }
+ 
++if (target_os == "android") {
++  if (v8_current_cpu == "arm" || v8_current_cpu == "x86") {
++    if (v8_enable_drumbrake) {
++      v8_enable_drumbrake = false
++    }
++  }
++}
++
+ if (v8_enable_drumbrake && v8_enable_webassembly) {
+   assert(
+       is_drumbrake_supported,
